app/services/preserver.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+class Preserver
+  # Provided as a mechanism for switching preservation strategies depending on
+  # the profile. For now, there's only one, so fall down to new.
+  def self.for(change_set:, change_set_persister:, storage_adapter: nil)
+    return NullPreserver unless change_set.try(:preserve?)
+    new(change_set: change_set, change_set_persister: change_set_persister, storage_adapter: storage_adapter)
+  end
+
+  class NullPreserver
+    def self.preserve!; end
+  end
+
+  attr_reader :change_set, :storage_adapter, :change_set_persister
+  delegate :resource, to: :change_set
+  def initialize(change_set:, storage_adapter: nil, change_set_persister:)
+    @change_set = change_set
+    @storage_adapter = storage_adapter || default_storage_adapter
+    @change_set_persister = change_set_persister
+  end
+
+  def preserve!
+    preserve_binary_content
+    # Don't preserve children unless this is the first time it's being
```

Comment:
Can you move this comment above the method? I totally missed the first line of the method when I read it the first time.